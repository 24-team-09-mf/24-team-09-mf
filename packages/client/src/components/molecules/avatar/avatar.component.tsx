// react
import { forwardRef, useState } from 'react'

// types
import { AvatarProps } from './avatar.types'

// styles
import { Wrapper, AvatarInput, AvatarImage } from './avatar.styles'

// images
import avatarDefault from '@/assets/images/avatarDefault.png'

export const Avatar = forwardRef<HTMLInputElement, AvatarProps>(
  ({ src, ...props }, ref) => {
    const [avatarSrc, setAvatarSrc] = useState(src || avatarDefault)

    const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
      const file = e.target.files?.[0]
      if (file) {
        const reader = new FileReader()
        reader.readAsDataURL(file)
        reader.onload = () => {
          setAvatarSrc(reader.result as string)
        }
      }
    }

    return (
      <Wrapper>
        <AvatarImage
          src={avatarSrc}
          alt="Аватар"
        />
        <AvatarInput
          type="file"
          ref={ref}
          {...props}
          onChange={onChange}
        />
      </Wrapper>
    )
  }
)

Avatar.displayName = 'Avatar'
