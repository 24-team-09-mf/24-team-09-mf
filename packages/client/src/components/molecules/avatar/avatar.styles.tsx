// lib
import styled from 'styled-components'

export const Wrapper = styled.div`
  position: relative;
  display: inline-block;
  margin-bottom: 20px;
  background-color: rgb(223 218 218);
  border-radius: 50%;
`

export const AvatarInput = styled.input`
  position: absolute;
  top: 0;
  left: 0;
  opacity: 0;
  width: 100%;
  height: 100%;
  cursor: pointer;
`
export const AvatarImage = styled.img`
  width: 120px;
  height: 120px;
  border-radius: 50%;
`
