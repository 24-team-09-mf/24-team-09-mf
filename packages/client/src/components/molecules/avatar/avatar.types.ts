import { HTMLAttributes } from 'react'

export interface AvatarProps extends HTMLAttributes<HTMLInputElement> {
  src?: string
}
