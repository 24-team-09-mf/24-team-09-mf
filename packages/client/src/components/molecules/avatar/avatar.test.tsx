import * as renderer from 'react-test-renderer'
import { Avatar } from './avatar.component'

describe('Тест компонента Avatar', () => {
  it('Тестирование снепшотом', () => {
    const component = renderer.create(<Avatar />).toJSON()
    expect(component).toMatchSnapshot()
  })
})
