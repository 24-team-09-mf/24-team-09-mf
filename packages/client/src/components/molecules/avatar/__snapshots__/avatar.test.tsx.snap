// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тест компонента Avatar Тестирование снепшотом 1`] = `
<div
  className="sc-beqWaB cdlOZE"
>
  <img
    alt="Аватар"
    className="sc-dmqHEX iErOfx"
  />
  <input
    accept=".png, .jpg"
    className="sc-gueYoa dJCDxn"
    onChange={[Function]}
    type="file"
  />
</div>
`;
